openshift_tools/ircbot/trello/trello.py

```diff
@@ -128,6 +150,97 @@ def get(self):
                                                  members)
         return results
 
+    def report(self, email=None, move=False):
+        """Generate reports"""
+        payload = self.report_payload()
+        print(payload)
+        _env_email = os.environ.get("trello_report_email_addresses", None)
+        if _env_email:
+            email = _env_email
+        if self.args:
+            if self.args.email:
+                email = self.args.email
+            if self.args.move:
+                move = self.args.move
+        week_number = date.today().isocalendar()[1]
+        if email:
+            subj = "OpenShift SRE Report for Week #{} (ending {})".format(
+                week_number, date.today().strftime("%d-%m-%Y"))
+            _board_url = "https://trello.com/b/{}".format(
+                os.environ.get("trello_board_id", None))
+            payload = "For more information visit the SRE 24x7 board {}\n\n{}".format(
+                _board_url, payload)
+            self.email_report(email, subj, payload)
+            print("Report emailed to {}".format(email))
+        if move:
```

Comment:
You have a lot of `if` blocks without `else` blocks. If you're not printing a reason for the `if` block, it would be hard to troubleshoot later why a specific command isn't working

Reply:
The non-print if blocks are really just to handle optional CLI args. Without a debug path I don't have a suggestion on improving this.